### PR TITLE
Added Support for ES6 Coverage

### LINKED ISSUE #97: Add code coverage reporting
I'm interested in including code coverage reporting within Slingshot. [Isparta](https://github.com/douglasduteil/isparta) and [nyc](https://github.com/bcoe/nyc) are two top candidates to consider.

### LINKED ISSUE #108: Add babel-based code coverage
Haven't gotten to try this out yet, but looks promising. http://dev.topheman.com/es6-code-coverage-with-babel-plugin/?utm_source=ESnextNews.com&utm_medium=Weekly%20Newsletter&utm_campaign=Week%2015

### PR DESCRIPTION
**Pull request by dwmkerr**
Hi Guys,

Here's a PR which adds coverage support.

### Badge Support

Coverage badge added to the README.md. If the PR is accepted, the coveralls project should be changed to one Cory sets up (instructions for this are in the FAQ).

image

### Optional integration with Coveralls

Support for the free-for-open-source coveralls tool, just check the FAQ.

image

### Coverage commands

Just run `npm run test:cover` to generate a local HTML coverage report, open it up with `open ./coverage/index.html`.

Would love to know your thoughts. Fixes #108. Fixes #97.

### Questions

Should we keep the coveralls support on by default, or have the users opt-in?

### CODE REVIEW

**Comment by coryhouse** [comment]
This looks great! I'm at a conference so I'll need to look this over a bit later, but my only initial concern is that Isparta isn't actively maintained. I believe there are some good alternatives to consider.

Thanks so much!

**Comment by coryhouse** [comment]
One typo: "report is writtent to"

Also, I'm confused this reference:
# Uncomment the line below if you have setup Coveralls as described in the FAQ.

The line is already uncommented in the commit, and would have to be uncommented for coveralls to work for us, right?

**Comment by dwmkerr** [comment]
[babel-plugin-__coverage__](https://github.com/dtinth/babel-plugin-__coverage__) looks like it may fit the bill, although not yet particularly mature. There's a suggestion in the repo that these capabilities will come natively to istanbul. The 'uncomment' line is indeed confusing, I'll remove all of that

**Comment by dwmkerr** [comment]
Although would you prefer the default for people to opt-in to coverage or opt-out?

**Comment by coryhouse** [comment]
Can you clarify the impact of opt-in by default?

File: package.json [hunk 1]
```diff
@@ -17,6 +17,8 @@
     "prebuild": "npm run clean-dist && npm run build:html && npm run lint && npm run test",
     "build": "babel-node tools/build.js && npm run open:dist",
     "test": "mocha tools/testSetup.js \"src/**/*.spec.js\" --reporter progress",
+    "test:cover": "babel-node $(npm bin)/isparta cover --root src --report html node_modules/mocha/bin/_mocha -- --require ./tools/testSetup.js \"src/**/*.spec.js\" --reporter progress",
+    "test:cover:travis": "babel-node ./node_modules/.bin/isparta cover --root src --report lcovonly node_modules/mocha/bin/_mocha -- --require ./tools/testSetup.js \"src/**/*.spec.js\" && cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js",
```
Comment (rndstr) [inline package.json]: `./node_modules/.bin` is automatically added to $PATH for these scripts, `babel-node isparta …` should be enough. Same for `test:cover`

**Comment by coryhouse** [comment]
@dwmkerr - Can you clarify the impact of opt-in by default?

**Comment by dwmkerr** [comment]
@coryhouse yes indeed, basically it means that if a user runs the build in travis, it will also attempt to then send the code coverage data (if the build is successful). If the user doesn't have a coveralls account it will fail to do so, let me get a quick screenshot of what that would look like

**Comment by dwmkerr** [comment]
OK so even without coveralls enabled the build passes successfully (it just doesn't generate a report) so I think it's good to go if you want it, I've updated the documentation just now too

**Comment by coryhouse** [comment]
@dwmkerr Thanks so much! 👍

**Comment by coryhouse** [comment]
@dwmkerr FYI - I just committed a few tweaks to provide Windows support. Windows didn't like this:

$(npm bin)

I also removed some unnecessary explicit paths to node_modules, and fixed a few typos. 

Thanks again! :)

**Comment by dwmkerr** [comment]
@coryhouse great thanks for making the tweaks, I didn't have windows environment to test on!

**Pull request merged**